### PR TITLE
Implemented stereo playback in DrumThumper.

### PR DESCRIPTION
**Pull request by android-usb**
Gentlemen,
Please take a look. I plan to do some more documentation, but I think the logic is good.

### CODE REVIEW

File: samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp [hunk 1]
```diff
@@ -22,20 +22,32 @@
...
-            outBuff[index] += data[mCurFrameIndex++];
+        if (numChannels == 1) {
+            // MONO output
+            for (int32_t frameIndex = 0; frameIndex < numWriteFrames; frameIndex++) {
+                outBuff[frameIndex] += data[mCurFrameIndex++];
+            }
+        } else if (numChannels == 2) {
+            // STEREO output
+            int dstSampleIndex = 0;
+            float leftGain = -(-mPan + 1.0) / 2.0f;
```
Comment (philburk) [inline samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp:45]: Why is leftGain negative? That could cause phase cancellation when centered.
How about:

    float rightGain = (mPan * 0.5) + 0.5;
    float leftGain = 1.0 - leftGain;

Also check out "Pan Laws" on this page:
http://www.cs.cmu.edu/~music/icm-online/readings/panlaws/

File: samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp [hunk 2]
```diff
@@ -22,20 +22,32 @@
...
+void OneShotSampleSource::mixAudio(float* outBuff, int numChannels, int32_t numFrames) {
     int32_t numSampleFrames = mSampleBuffer->getNumSampleFrames();
     int32_t numWriteFrames = mIsPlaying
                          ? std::min(numFrames, numSampleFrames - mCurFrameIndex)
                          : 0;
 
     if (numWriteFrames != 0) {
         // Mix in the samples
-        int32_t lastIndex = mCurFrameIndex + numWriteFrames;
+        // int32_t lastIndex = mCurFrameIndex + numWriteFrames;
```
[the remark this answers is not in the document]
Comment (android-usb) [reply samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp:33]: Done.

File: samples/iolib/src/main/cpp/player/SampleSource.h [hunk 3]
```diff
@@ -28,24 +28,43 @@ namespace iolib {
...
-    SampleSource(SampleBuffer *sampleBuffer)
-     : mSampleBuffer(sampleBuffer), mCurFrameIndex(0), mIsPlaying(false) {};
+    // Pan position of the audio in a stereo mix
+    // [left:-1.0f] <- [center: 0.0f] -> -[right: 1.0f]
+    static constexpr float PAN_HARDLEFT = -1.0f;
+    static constexpr float PAN_HARDRIGHT = 1.0f;
+    static constexpr float PAN_CENTER = 0.0f;
+
+    SampleSource(SampleBuffer *sampleBuffer, float pan)
+     : mSampleBuffer(sampleBuffer), mCurFrameIndex(0), mIsPlaying(false), mPan(pan) {};
```
[the remark this answers is not in the document]
Comment (android-usb) [reply samples/iolib/src/main/cpp/player/SampleSource.h:42]: Good call. Done.

File: samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp (hunk 1 above)
Comment (android-usb) [reply samples/iolib/src/main/cpp/player/OneShotSampleSource.cpp:45]: Clumsy arithmetic on my part. Done.

**Pull request merged**